components/ModelLicense/licenseUtils.ts

```diff
@@ -0,0 +1,175 @@
+import type { CharacterModelLicenseSerializer } from '@/types/Response';
+import type { VRM1Meta } from '@pixiv/three-vrm';
+
+type LicenseProperty = keyof CharacterModelLicenseSerializer;
+
+/**
+ * VRoid Hubで設定出来るVRM0のライセンスから表示テキストを返す
+ * @param license
+ * @param property
+ * @returns
+ */
+export const getHubVrm0LicenseValue = (license: CharacterModelLicenseSerializer, property: LicenseProperty): string => {
+  const status = license[property];
+
+  if (status === 'default') return '未設定';
+
+  switch (property) {
+    case 'characterization_allowed_user':
+      return status === 'everyone' ? 'OK' : 'NG';
+    case 'personal_commercial_use':
+      if (status === 'profit') return 'OK';
+      if (status === 'nonprofit') return '非営利のみ';
+      return 'NG';
+    case 'credit':
+      return status === 'necessary' ? '必要' : '不要';
+    case 'modification':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'redistribution':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'sexual_expression':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'violent_expression':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'corporate_commercial_use':
+      return status === 'allow' ? 'OK' : 'NG';
+    default:
+      return status === 'allow' ? 'OK' : 'NG';
+  }
+};
+
+/**
+ * VRoid Hubで設定出来るVRM0のライセンスからテキストの表示色の種類を返す
+ * @param license
+ * @param property
+ * @returns
+ */
+export const getHubVrm0LicenseVariant = (
+  license: CharacterModelLicenseSerializer,
+  property: LicenseProperty,
+): 'default' | 'allow' | 'bold' => {
+  const status = license[property];
+
+  switch (property) {
+    case 'characterization_allowed_user':
+      return status === 'everyone' ? 'allow' : 'default';
+    case 'personal_commercial_use':
+      return status === 'disallow' || status === 'default' ? 'default' : 'allow';
+    case 'credit':
+      return status === 'necessary' ? 'bold' : 'default';
+    case 'modification':
+      return status === 'allow' ? 'allow' : 'default';
+    case 'redistribution':
+      return status === 'allow' ? 'allow' : 'default';
+    case 'sexual_expression':
+      return status === 'allow' ? 'allow' : 'default';
+    case 'violent_expression':
+      return status === 'allow' ? 'allow' : 'default';
+    case 'corporate_commercial_use':
+      return status === 'allow' ? 'allow' : 'default';
+    default:
+      return status === 'allow' ? 'allow' : 'default';
+  }
+};
+
+type vrm1MetaProperty = keyof VRM1Meta | 'personalCommercialUse' | 'modificationRedistribution';
+
+/**
+ * VRM1.0のライセンスから表示テキストを返す
+ * @param vrm1Meta
+ * @param property
+ * @returns
+ */
+export const getVrm1LicenseValue = (vrm1Meta: VRM1Meta, property: vrm1MetaProperty) => {
+  //「 個人の商用利用」は法人利用とプロパティが共用のため、commercialUsageの値を参照し判定する
+  if (property === 'personalCommercialUse') {
+    const status = vrm1Meta.commercialUsage;
+    if (typeof status === 'undefined') return '未設定';
```

Comment:
ここは文字列でなく、undefinedだった場合の想定です。

```suggestion
    if (typeof status === undefined) return '未設定';
```

three-vrmのVRM1Metaの定義上もundefinedを許容する型定義になっています

https://github.com/pixiv/three-vrm/blob/9a1814ade92e2617cb4b96e6210e15e45e2ff935/packages/three-vrm-core/src/meta/VRM1Meta.ts#L74C3-L74C18

Reply:
typeof演算子を使っているので一応意図通りにはなっているはずです。
ただ、undefined | false | trueのプロパティがあったのでこのように実装しましたが、`status == null`の方が良かったかもしれないです。

nullを区別したい項目はないですよね？

Reply:
失礼しました。確かにtypeofがついていました。

nullを区別したい項目はないですね。

Reply:
https://github.com/pixiv/VRoidHub-API-Example/pull/7/commits/7319d0476ce631ea52d6e8bc2c1ba69efacbb7d0 にて`status == null`へ変更しました。